lib/vuln.js

```diff
@@ -94,10 +94,16 @@ class Vuln {
   }
 
   toJSON () {
-    // sort so that they're always in a consistent order
+    const edgesIn = Array.from(this.nodes.values()).flatMap(a => Array.from(a.edgesIn.values())).map(a => a.toJSON())
+    const types = new Set(edgesIn.map(a => a.type).sort())
+    const isDirect = edgesIn.any(edge => edge.from === '')
```

Comment:
```suggestion
    const isDirect = [...this.nodes.values()].some(n => [...n.edgesIn].some(e => e.from.isProjectRoot()))
```

Reply:
That's the function I knew must exist but didn't find! Thanks